### DISCUSSION TITLE
srsRAN + FlexRIC - KPM Formats

### DISCUSSION THREAD

**Discussion by mas-oliveira** [srsran/srsRAN_Project#437] [General Help] [answered]
Hello, y'all.

I'm trying to make a data collection using the new tutorial update of the srsRAN that uses the "br-flexric" commit of the Mosaic5G project. At the moment, I am using the KPM Format 1, but there is a problem I can't get the measure names in the sm_cb_kpm.

On the function **gen_act_def_frmt_1** I have made this cycle:
for (size_t i = 0; i < count_aux; i++) {
    printf("---%s \n", dst.meas_info_lst[i].meas_type.name.buf);
}

While running I could verify that the actions are being to the kpm action format, but when I receive the messages I can't get that information.

But then when I receive KPM Format 1 subscription messages they're always like this:

Size of message info list = 0

RECEIVED INT MEAS value = 18

Size of message info list = 0

RECEIVED INT MEAS value = 34387

Size of message info list = 0

RECEIVED INT MEAS value = 0

Size of message info list = 0

RECEIVED INT MEAS value = 18

Size of message info list = 0

RECEIVED INT MEAS value = 34282

Where the size of the message info list is according to the value of this variable: **msg_frm_1->meas_info_lst_len** in the **sm_cb_kpm** function.

Since I have this metrics:  ( { name = "DRB.UEThpDl" }, { name = "DRB.UEThpUl" }, { name = "DRB.RlcPacketDropRateDl" }, { name = "DRB.RlcSduTransmittedVolumeDL"}, { name = "DRB.RlcSduTransmittedVolumeUL"})

I think that this is always by the order that I have in my conf file. But, does someone know why the message doesn't carry any information with the measures' names?

By the way, someone is trying to get the gNB to accept format 4 or any other formats of the KPMs?

**Comment by yc541** [comment, marked as answer]
Looking at the RIC indication messages in pcap, the names of the requested measurements are visible in the pm containers. So it's up to the xapp to extract the names, or like you mentioned do a workaround and get the names from the config file. 

I couldn't get the gNB to work with format 4 either. It seems that the action definition format is not decoded correctly at the gNB. In the code block e2_sm_kpm_action_definition_s::action_definition_formats_c_::unpack(cbit_ref& bref), the line e.unpack(bref) decodes it to 2 rather than 4. In the caller function e2_sm_kpm_action_definition_s::unpack(cbit_ref& bref), ric_style_type is correctly decoded to 4 though.

**Reply by mas-oliveira** [reply]
Thanks @yc541 for the answer. I have to make a better check about the populated structures when the message arrives then! 

I was checking that too, I'll see if I can do anything about the formats that can't be decode at gNB. 

Thanks again

**Comment by yagoda** [comment]
Hi @mas-oliveira, 

In preparation for our upcoming workshop in DC, we are contacting some users of our E2 solution. Can you please share a few lines about your experience with the E2 interface, what you were trying to achieve, and how it worked out for you?

Kind Regards,
Justin
